x/rewards/client/cli/query_generate_ibc_iprpc_tx.go

```diff
@@ -0,0 +1,213 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	ibctransfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	ibcchanneltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/x/rewards/types"
+	"github.com/spf13/cobra"
+)
+
+var _ = strconv.Itoa(0)
+
+const (
+	FlagMemoOnly = "memo-only"
+)
+
+func CmdQueryGenerateIbcIprpcTx() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "generate-ibc-iprpc-tx [spec] [duration] [amount] [src-port] [src-channel] --from <alice> --node <node-URI>",
+		Short: `Query for generating an ibc-transfer TX JSON for IPRPC funding over IBC. 
+		The generated TX is a regular ibc-transfer TX with a custom memo. The memo holds the necessary information for a IPRPC over IBC request: 
+		creator, spec ID the user wishes to fund, and duration of the fund in months.
+		Note, the memo's spec ID field is not validated. A valid spec ID is one that is registered on-chain in the Lava blockchain. To 
+		check whether a spec ID is valid, use the "lavad q spec show-chain-info" query.
+		
+		The query must be called with the following flags:
+		 - "--from" flag which will be the ibc-transfer sender. Note, the TX should be signed by the sender.
+		 - "--node" flag which should be the node from which the ibc-transfer TX originates (probably not a Lava node).
+		
+		To generate only the memo, so it can be embedded manually when sending an ibc-transfer, use the optional 
+		--memo-only flag (in this case, the amount, src-port, and src-channel arguments will be ignored).`,
+		Args: cobra.RangeArgs(2, 5),
+		Example: `
+		lavad q rewards generate-ibc-iprpc-tx ETH1 3 100ulava transfer channel-0 --from bob --node tcp://localhost:36657 > tx.json
+		lavad tx sign tx.json --from bob > signedTx.json
+		lavad tx broadcast signedTx.json --from bob --node tcp://localhost:36657
+		`,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			spec := args[0]
+			duration := args[1]
+
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			clientCtx.GenerateOnly = true
+			sender := clientCtx.FromAddress.String()
+
+			// create memo
+			senderAddr, err := utils.ParseCLIAddress(clientCtx, sender)
+			if err != nil {
+				return err
+			}
+			memo, err := createIbcIprpcMemo(senderAddr, spec, duration)
```

Comment:
what about the amount?

Reply:
the amount is not part of the memo. It's an argument to the ibc-transfer TX